packages/@aws-cdk/aws-codecommit/lib/repository.ts

```diff
@@ -53,6 +55,14 @@ export abstract class RepositoryRef extends cdk.Construct {
         };
     }
 
+    public addToPipelineStage(stage: actions.IStage, name: string, props: CommonPipelineSourceProps): PipelineSource {
+        return new PipelineSource(this.parent!!, name, {
```

Comment:
I don't reckon you need two of the `!`s :)

Reply:
Why attach it to that `parent` and not to `this`?

Reply:
> I don't reckon you need two of the !s :)

Yep, my bad. Will fix.

Reply:
> Why attach it to that parent and not to this?

Because I want to make this:

```ts
repository.addToPipeline(stage, 'CodeCommit', {
    artifactName: 'SourceOutput',
});
```

be equivalent to this:

```ts
new codecommit.PipelineSourceAction(this, 'CodeCommit', {
    stage: stage,
    artifactName: 'SourceOutput',
    repository: repository,
});
```

If we attach the Action to the `Repository`, instead of its parent, that first call above would be equivalent to:

```ts
new codecommit.PipelineSourceAction(repository, 'CodeCommit', {
    // rest is the same as above...
});
```
, which is not the same thing.

Reply:
Why is it important that they will be attached to the same construct?

Reply:
It's not strictly required. But, imagine a world where we move to managing a Role per each Action (instead of having a big Pipeline Role with all of the permissions). If an Action will create that Role, then the code `repository.addToPipeline(stage, 'CodeCommit', { ...})` and `new codecommit.PipelineSourceAction(this, 'CodeCommit', { repository, ... })` will produce different CloudFormation templates (the paths will be different, and so the logical names assigned to the Role will be different). Using the parent here prevents that.

What's a good reason for _not_ using the parent here?